Sources/Layout/MessagesCollectionViewFlowLayout.swift

```diff
@@ -45,31 +44,51 @@ open class MessagesCollectionViewFlowLayout: UICollectionViewFlowLayout {
     open var messageLabelFont: UIFont {
         didSet {
             emojiLabelFont = messageLabelFont.withSize(2 * messageLabelFont.pointSize)
+            textMessageAttributesCache.removeAllObjects()
+            attributedTextMessageAttributesCache.removeAllObjects()
+            emojiMessageAttributesCache.removeAllObjects()
         }
     }
 
-    /// The font to be used by `TextMessageCell` for `MessageData.emoji(String)` case.
-    ///
-    /// Note: The default value of this property is 2x the `messageLabelFont`.
     internal var emojiLabelFont: UIFont
 
-    /// Determines the maximum number of `MessageCollectionViewCell` attributes to cache.
-    ///
-    /// Note: The default value of this property is 500.
-    open var attributesCacheMaxSize: Int = 500 {
-        didSet {
-            layoutContextCache.countLimit = attributesCacheMaxSize
-        }
+    // MARK: - Layout
+
+    open var textMessageLayout = TextMessageLayout() {
+        didSet { textMessageAttributesCache.removeAllObjects() }
+    }
```

Comment:
Not sure if I want these to be open to be honest, I don't want users removing the cache clearing behavior

Reply:
We could potentially put the properties of `TextMessageLayout` and the cache into a class called `TextMessageSizeCalculator` -- similar to JSQ's `MessageBubbleSizeCalculator`. Then we can also store underlying methods in here such as `messageContainerSize` etc.